### PR TITLE
fix headers RecoMuon pkgs

### PR DESCRIPTION
**Pull request by mrodozov**
Fix headers related to #15248

### CODE REVIEW

File: RecoMuon/MuonIdentification/plugins/MuonRefProducer.h [hunk 1]
```diff
@@ -21,6 +21,7 @@
 #include "FWCore/Framework/interface/global/EDProducer.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "DataFormats/MuonReco/interface/Muon.h"
+#include "DataFormats/MuonReco/interface/MuonFwd.h"
```
Comment (perrotta) [inline RecoMuon/MuonIdentification/plugins/MuonRefProducer.h]: Muon.h and MuonFwd.h are repeated in the MuonRefProducer.cc
Would you like to clean up the MuonRefProducer.cc implementation file before merging this PR?

**Comment by perrotta** [comment]
@mrodozov : please let us know if you plan to clean up the implementation file as suggested in https://github.com/cms-sw/cmssw/pull/23385#pullrequestreview-124811459, or if you prefere to let things as they are now instead

**Comment by mrodozov** [comment]
please test

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass

**Comment by Martin-Grunewald** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**